packages/kit/src/core/adapt/prerender.js

```diff
@@ -97,7 +97,7 @@ export async function prerender({ cwd, out, log, config, build_data, fallback, a
 	 */
 	function normalize(path) {
 		if (config.kit.trailingSlash === 'always') {
-			return path.endsWith('/') ? path : `${path}/`;
+			return path.endsWith('/') || path.endsWith('.html') ? path : `${path}/`;
```

Comment:
This no longer does what I would expect for an option named `always`, which would be to always do it.

This sort of feels like it should accept a function to let the user decide like https://github.com/sveltejs/kit/pull/2008 and https://github.com/sveltejs/kit/pull/2007. Then you could have it do whatever you want

Reply:
Svelte files are designed to compose components made with HTML/CSS/JS so users won't generate files other than `.html` file, I think.

But endpoints like `sitemap.xml.js` are considered, so I think this option should be changed, as you say.

Can I change this option to something like `never | ignore | (path: string) => boolean` , in this PR?  Or do you think it would be better to change it as a separate PR?

Reply:
Yes, you're right that you'd use an endpoint

This probably shouldn't be a function after all. It gets stringified during build and could be confusing to users

I think a better logic here might be to do it only if the last path segment does not contain `.`, but let me ask the other maintainers what they think

Reply:
Hmm, that might not be a great solution either. E.g. that wouldn't work with the URL https://bundlephobia.com/package/svelte@3.42.1

I think we just leave this as is and say you should not use the option `always` if you're generating `.html` files